Bottom: d3934cdbe34b58aa8ab19f5779a618d2a570bea4
Top:    a6f89561d3da2a97815426b08127647dbc9260ac
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2012-03-25 17:43:59 -0400

Refresh of i2c-resource

---

diff --git a/arch/arm/boot/dts/ea3131.dts b/arch/arm/boot/dts/ea3131.dts
index 82dffcd..7c7cbfc 100644
--- a/arch/arm/boot/dts/ea3131.dts
+++ b/arch/arm/boot/dts/ea3131.dts
@@ -35,10 +35,12 @@
 		spi@15002000 {
 			at45db321d@0 {
 				compatible = "atmel,at45db321d", "atmel,at45", "atmel,dataflash";
+				spi-max-frequency = <1000000>;
 				reg = <0>;
 			};
 			ads7846@1 {
 				compatible = "ti,ads7846";
+				spi-max-frequency = <1200000>;
 				reg = <1>;
 				interrupt = <37>;
 			};
diff --git a/drivers/input/touchscreen/ads7846.c b/drivers/input/touchscreen/ads7846.c
index 23fd901..9ac17b0 100644
--- a/drivers/input/touchscreen/ads7846.c
+++ b/drivers/input/touchscreen/ads7846.c
@@ -1423,11 +1423,22 @@ static int __devexit ads7846_remove(struct spi_device *spi)
 	return 0;
 }
 
+#if defined(CONFIG_OF)
+static const struct of_device_id ads7846_of_match[] = {
+	{ .compatible = "ti,ads7846" },
+	{},
+};
+MODULE_DEVICE_TABLE(of, ads7846_of_match);
+#endif
+
 static struct spi_driver ads7846_driver = {
 	.driver = {
 		.name	= "ads7846",
 		.owner	= THIS_MODULE,
 		.pm	= &ads7846_pm,
+#ifdef CONFIG_OF
+		.of_match_table = ads7846_of_match,
+#endif
 	},
 	.probe		= ads7846_probe,
 	.remove		= __devexit_p(ads7846_remove),
diff --git a/drivers/of/of_spi.c b/drivers/of/of_spi.c
index 6dbc074..4317938 100644
--- a/drivers/of/of_spi.c
+++ b/drivers/of/of_spi.c
@@ -6,6 +6,8 @@
  * tree.
  */
 
+#define DEBUG
+
 #include <linux/module.h>
 #include <linux/of.h>
 #include <linux/device.h>
diff --git a/drivers/spi/spi_lpc313x.c b/drivers/spi/spi_lpc313x.c
index 7e296ef..0157806 100644
--- a/drivers/spi/spi_lpc313x.c
+++ b/drivers/spi/spi_lpc313x.c
@@ -931,6 +931,9 @@ static int __init lpc313x_spi_probe(struct platform_device *pdev)
 		ret = -EBUSY;
 		goto errout4;
 	}
+#ifdef CONFIG_OF
+	master->dev.of_node = of_node_get(pdev->dev.of_node);
+#endif
 
 	ret = spi_register_master(master);
 	if (ret)
diff --git a/dt_config b/dt_config
index d4f683b..825ff36 100644
--- a/dt_config
+++ b/dt_config
@@ -909,7 +909,39 @@ CONFIG_INPUT_MOUSEDEV_SCREEN_Y=768
 # CONFIG_INPUT_MOUSE is not set
 # CONFIG_INPUT_JOYSTICK is not set
 # CONFIG_INPUT_TABLET is not set
-# CONFIG_INPUT_TOUCHSCREEN is not set
+CONFIG_INPUT_TOUCHSCREEN=y
+CONFIG_TOUCHSCREEN_ADS7846=y
+# CONFIG_TOUCHSCREEN_AD7877 is not set
+# CONFIG_TOUCHSCREEN_AD7879 is not set
+# CONFIG_TOUCHSCREEN_ATMEL_MXT is not set
+# CONFIG_TOUCHSCREEN_AUO_PIXCIR is not set
+# CONFIG_TOUCHSCREEN_BU21013 is not set
+# CONFIG_TOUCHSCREEN_CY8CTMG110 is not set
+# CONFIG_TOUCHSCREEN_DYNAPRO is not set
+# CONFIG_TOUCHSCREEN_HAMPSHIRE is not set
+# CONFIG_TOUCHSCREEN_EETI is not set
+# CONFIG_TOUCHSCREEN_EGALAX is not set
+# CONFIG_TOUCHSCREEN_FUJITSU is not set
+# CONFIG_TOUCHSCREEN_GUNZE is not set
+# CONFIG_TOUCHSCREEN_ELO is not set
+# CONFIG_TOUCHSCREEN_WACOM_W8001 is not set
+# CONFIG_TOUCHSCREEN_MAX11801 is not set
+# CONFIG_TOUCHSCREEN_MCS5000 is not set
+# CONFIG_TOUCHSCREEN_MTOUCH is not set
+# CONFIG_TOUCHSCREEN_INEXIO is not set
+# CONFIG_TOUCHSCREEN_MK712 is not set
+# CONFIG_TOUCHSCREEN_PENMOUNT is not set
+# CONFIG_TOUCHSCREEN_TOUCHRIGHT is not set
+# CONFIG_TOUCHSCREEN_TOUCHWIN is not set
+# CONFIG_TOUCHSCREEN_PIXCIR is not set
+# CONFIG_TOUCHSCREEN_USB_COMPOSITE is not set
+# CONFIG_TOUCHSCREEN_TOUCHIT213 is not set
+# CONFIG_TOUCHSCREEN_TSC_SERIO is not set
+# CONFIG_TOUCHSCREEN_TSC2005 is not set
+# CONFIG_TOUCHSCREEN_TSC2007 is not set
+# CONFIG_TOUCHSCREEN_W90X900 is not set
+# CONFIG_TOUCHSCREEN_ST1232 is not set
+# CONFIG_TOUCHSCREEN_TPS6507X is not set
 # CONFIG_INPUT_MISC is not set
 
 #
diff --git a/sound/soc/lpc313x/lpc313x-i2s.c b/sound/soc/lpc313x/lpc313x-i2s.c
index e180788..d255fa7 100644
--- a/sound/soc/lpc313x/lpc313x-i2s.c
+++ b/sound/soc/lpc313x/lpc313x-i2s.c
@@ -398,7 +398,6 @@ EXPORT_SYMBOL_GPL(lpc313x_i2s_dai);
 
 static __devinit int lpc313x_i2s_dev_probe(struct platform_device *pdev)
 {
-	printk("JDS = lpc313x_i2s_dev_probe\n");
 	return snd_soc_register_dai(&pdev->dev, &lpc313x_i2s_dai);
 }
